### DISCUSSION TITLE
Can't connect to Todoist servers when in Class Library(.NET Framework) project

### DISCUSSION THREAD

**Discussion by matz33** [olsh/todoist-net#27] [Q&A] [answered]
Hi,

Your package is running great when added through Nugets on a Visual Basic Forms project. I can connect to my Todoist, and retrieve list of projects for exemple.

When running the same piece of code to retrieve list of project inside a "Class Library(.NET Framework)", I have 2 errors to connect to Todoist servers, when using GetResourcesAsync method : 
1. WebException: Unable to connect to the remote server
2. SocketException: An attempt was made to access a socket in a way forbidden by its access permissions

I uninstalled your package in Nugets, downloaded the source code from Github, added the Todoist.Net project to my solution, added a reference to it, in order to find from where the issues are. I found that this is in the file TodoistRestClient.cs, line78 in the method PostAsync. 

My Class is using Framework 4.8, I upgrading yours that seems to be 4.5 to 4.8, same result.

I've seen that you have just upgrading your project to version 5, same result.

Find attached all screenshots I could do to show the error.

Can you help ?

First exception (in French, I translated it) :
![image](https://user-images.githubusercontent.com/68858708/229368184-cd79af8c-d3bc-4832-9e5d-30ab691d4b7c.png)

Second exception :
![image](https://user-images.githubusercontent.com/68858708/229368199-20f6912b-71e4-4781-8547-188cd3f72e1f.png)

Exception with your source code :
![image](https://user-images.githubusercontent.com/68858708/229368222-9071e5fe-a193-49cb-a7bb-8e5fd23ac722.png)

My project settings : 
![image](https://user-images.githubusercontent.com/68858708/229368239-ca2eb369-fc5b-4885-bdc4-3355e31baa26.png)

Upgraded your project to 4.8 (but same error) :
![image](https://user-images.githubusercontent.com/68858708/229368270-314478f9-0ea6-48cc-8e81-06bdfcfcfead.png)
![image](https://user-images.githubusercontent.com/68858708/229368302-a200ad96-c125-4c74-9b72-7bc337e09150.png)

**Comment by olsh** [comment]
I don't think that the network issues are related to the library.
Also, it's hard to understand what exactly blocks the connection.

**Reply by matz33** [reply, marked as answer]
Thanks a lot for this quick answer.

My problem is that same code in a Windows Forms project with .NET 6.0 works with no issues while the one with .Net Class using Framework 4.8 does not, on the same computer, on the same time. This project within this framework is to build a DLL that is called in another application, using COM. 

I tried with another project with .Net class using framework 4.8, but not related to this COM/application, and it worked, So I guess the issue is because of this application and/or COM environment that is making some network problems
